lib/screens/forgot_password/cubit/forgot_password_cubit.dart

```diff
@@ -12,37 +12,40 @@ class ForgotPasswordCubit extends Cubit<ForgotPasswordState> {
 
   ForgotPasswordCubit(this._forgotPasswordRepository) : super(ForgotEmailInitial());
 
-  void sendOTP(String email) async {
+  void sendOTP(String email, ForgotPasswordState state) async {
     try {
       emit(ForgotLoading());
       await _forgotPasswordRepository.sendOTP(email);
       emit(ForgotOTPInitial());
     } on NetworkException {
-      emit(ForgotNetworkError(S.networkException));
+      emit(ForgotNetworkError(S.networkException, state));
+    } on ResponseException catch (e) {
+      emit(ForgotOTPFailure(e.message, state));
     }
   }
 
-  Future<void> verifyOTP(String a) async {
+  Future<void> verifyOTP(String a, ForgotPasswordState state) async {
```

Comment:
1. give better variable names.
2. as per the API requirements, this function needs String otp and String email.
3. to prevent confusions, name it `checkOTP` as `check_otp` is the api endpoint.